src/screens/NanoContract/NanoContractRegisterScreen.js

```diff
@@ -0,0 +1,323 @@
+import { TextValue } from '../../components/TextValue';
+import { COLORS } from '../../styles/themes';
+import {
+  firstAddressRequest,
+  nanoContractRegisterReady,
+  nanoContractRegisterRequest
+} from '../../actions';
+import {
+  feedbackSucceedText,
+  hasFailed,
+  hasSucceeded,
+  isLoading
+} from './helper';
+import Spinner from '../../components/Spinner';
+import FeedbackModal from '../../components/FeedbackModal';
+import errorIcon from '../../assets/images/icErrorBig.png';
+import checkIcon from '../../assets/images/icCheckBig.png';
+import { FeedbackContent } from '../../components/FeedbackContent';
+import { hasError } from '../../utils';
+
+/**
+ * Validates the formModel, returning the invalidModel.
+ * If there is no error in the formModel, the invalidModel is returned empty.
+ */
+function validate(formModel) {
+  const invalidModel = {};
+
+  if (!formModel.ncId) {
+    invalidModel.ncId = t`Nano Contract ID is required.`;
+  }
+
+  return invalidModel;
+}
+
+export function NanoContractRegisterScreen({ navigation }) {
+  const dispatch = useDispatch();
+  const { address, error } = useSelector((state) => state.firstAddress);
+  const registerState = useSelector((state) => ({
+    registerStatus: state.nanoContract.registerStatus,
+    registerFailureMessage: state.nanoContract.registerFailureMessage,
+  }));
+
+  const [isClean, setClean] = useState(true);
+  const [formModel, setFormModel] = useState({
+    ncId: null,
+  });
+  const [invalidModel, setInvalidModel] = useState({
+    ncId: null,
+  });
+
+  /* @param {
+   *   'nodeUrl'|'explorerUrl'|'explorerServiceUrl'|'walletServiceUrl'|'walletServiceWsUrl'
+   * } name */
+  const handleInputChange = useCallback(
+    (name) => (value) => {
+      if (isClean) {
+        setClean(false);
+      }
+
+      // update invalid model
+      const invalidModelCopy = { ...invalidModel };
+      delete invalidModelCopy[name];
+      setInvalidModel(invalidModelCopy);
+
+      // update form model
+      const form = {
+        ...formModel,
+        [name]: value,
+      };
+      setFormModel(form);
+
+      // validate form model and update invalid model
+      setInvalidModel(validate(form));
+    },
+    [isClean, invalidModel, formModel]
+  );
+
+  const handleSubmit = useCallback(
+    () => {
+      const newInvalidModel = validate(formModel);
+      if (hasError(newInvalidModel)) {
+        setInvalidModel(newInvalidModel);
+        return;
+      }
+
+      const { ncId } = formModel;
+      dispatch(nanoContractRegisterRequest({ address, ncId }));
+    },
+    [formModel]
+  );
+
+  const handleFeedbackModalDismiss = () => {
+    dispatch(nanoContractRegisterReady());
+  };
+
+  const navigatesToNanoContractTransactions = () => {
+    dispatch(nanoContractRegisterReady());
+    navigation.replace('NanoContractDetailsScreen', { ncId: formModel.ncId });
+  };
+
+  useEffect(() => {
+    dispatch(firstAddressRequest());
```

Comment:
I don't think it makes sense to execute this request every time you register a nano contract. This will always be the same for the wallet, so you can check if it's already saved in redux to execute it only once.